db/ssl.go

```diff
@@ -0,0 +1,170 @@
+package db
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/turbot/steampipe/constants"
+)
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		log.Println("File exists\n")
+		return true
+	}
+	log.Println("File does not exist\n")
+	return false
+
+}
+
+func SslMode() string {
+	certExists := fileExists(filepath.Join(getDataLocation(), constants.ServerCert))
+	privateKeyExists := fileExists(filepath.Join(getDataLocation(), constants.ServerKey))
+	if certExists && privateKeyExists {
+		return "require"
+	}
+	return "disable"
+}
+
+func SslStatus() string {
+	certExists := fileExists(filepath.Join(getDataLocation(), constants.ServerCert))
+	privateKeyExists := fileExists(filepath.Join(getDataLocation(), constants.ServerKey))
+	if certExists && privateKeyExists {
+		return "on"
+	}
+	return "off"
+}
+
+func writeCertFile(filePath string, cert string) error {
+	return ioutil.WriteFile(filePath, []byte(cert), 0600)
+}
+
+func generateSelfSignedCertificate() (err error) {
+
+	// Check if the file exists if the file exists then do not generate the cert and key
+	certExists := fileExists(filepath.Join(getDataLocation(), constants.ServerCert))
+	privateKeyExists := fileExists(filepath.Join(getDataLocation(), constants.ServerKey))
+
+	if certExists && privateKeyExists {
+		return nil
+	}
+
+	// Create your own certificate authority
+	caPriv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		log.Println("Private key creation failed")
+		return err
+	}
+
+	caInput := x509.Certificate{
+		SerialNumber: big.NewInt(2020),
+
+		Subject: pkix.Name{
+			Organization: []string{"Turbot"},
+			Country:      []string{"US"},
+			Province:     []string{""},
+			Locality:     []string{"New Jersey"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	cert, err := x509.CreateCertificate(rand.Reader, &caInput, &caInput, &caPriv.PublicKey, caPriv)
+
+	if err != nil {
+		log.Println("Failed to create certificate")
```

Comment:
do we need to print out this error - what happens to the error that is returned? Have you tested this to verify how the error is handled?